be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -463,13 +463,15 @@ void JsonReader::_reorder_column(std::vector<SlotDescriptor*>* slot_descs,
             continue;
         }
 
-        auto itr = std::find_if(slot_descs->begin(), slot_descs->end(),
-                                [key](const SlotDescriptor* desc) { return desc->col_name() == key; });
+        key_set.insert(key);
 
-        if (itr != slot_descs->end()) {
-            auto tmp = *itr;
-            *itr = slot_descs->at(idx);
-            slot_descs->at(idx) = tmp;
+        // Find the SlotDescriptor with the json document key.
+        auto itr = std::find_if(ordered_slot_descs.begin(), ordered_slot_descs.end(),
```

Comment:
Why not use a unordered_map<string, SlotDescriptor*> to accelerate this search?

Reply:
@imay Updated.